FRENCH/src/ch06-03-if-let.md

```diff
@@ -0,0 +1,279 @@
+    _ => count += 1,
+}
+```
+-->
+
+```rust
+# #[derive(Debug)]
+# enum USAState {
+#    Alabama,
+#    Alaska,
+# }
+#
+# enum USACoin {
+#    Penny,
+#    Nickel,
+#    Dime,
+#    Quarter(USAState),
+# }
+# let piece = USACoin::Penny;
+let mut compteur = 0;
+match piece {
+    USACoin::Quarter(etat) => println!("Il s'agit d'un Quarter de l'état de {:?} !", etat),
+    _ => compteur += 1,
+}
+```
+
+<!--
+Or we could use an `if let` and `else` expression like this:
+-->
+
+Ou nous pourrions utiliser des expressions `if let` et `else` comme ceci :
+
+<!--
+```rust
+# #[derive(Debug)]
+# enum UsState {
+#    Alabama,
+#    Alaska,
+# }
+#
+# enum Coin {
+#    Penny,
+#    Nickel,
+#    Dime,
+#    Quarter(UsState),
+# }
+# let coin = Coin::Penny;
+let mut count = 0;
+if let Coin::Quarter(state) = coin {
+    println!("State quarter from {:?}!", state);
+} else {
+    count += 1;
+}
+```
+-->
+
+```rust
+# #[derive(Debug)]
+# enum USAState {
+#    Alabama,
+#    Alaska,
+# }
+#
+# enum USACoin {
+#    Penny,
+#    Nickel,
+#    Dime,
+#    Quarter(USAState),
+# }
+# let piece = USACoin::Penny;
+let mut compteur = 0;
+if let USACoin::Quarter(etat) = piece {
+    println!("Il s'agit d'un Quarter de l'état de {:?}!", etat);
+} else {
+    compteur += 1;
+}
+```
+
+<!--
+If you have a situation in which your program has logic that is too verbose to
+express using a `match`, remember that `if let` is in your Rust toolbox as well.
+-->
+
+Si vous vous trouvez dans une situation dans laquelle votre programme a une
+logique qui est trop verbeuse en utilisant un `match`, souvenez-vous que
+`if let` est aussi dans votre boite à outils Rust.
+
+<!--
+## Summary
+-->
+
+## Résumé
+
+<!--
+We’ve now covered how to use enums to create custom types that can be one of a
+set of enumerated values. We’ve shown how the standard library’s `Option<T>`
+type helps you use the type system to prevent errors. When enum values have
+data inside them, you can use `match` or `if let` to extract and use those
+values, depending on how many cases you need to handle.
+-->
+
+Nous avons désormais appris comment utiliser les énumérations pour créer des
+types personnalisés qui peuvent faire partie d'un jeu de valeurs recensées. Nous
+avons montré comment le type `Option<T>` de la bibliothèque standard vous aide
+à utiliser le système de types pour éviter les erreurs. Lorsque les valeurs
+d'énumération ont une valeur à l'intérieur d'elles, vous pouvez utiliser `match`
+ou `if let` pour extraire et utiliser ces valeurs, à choisir en fonction du
+nombre de cas que vous voulez gérer.
+
+<!--
+Your Rust programs can now express concepts in your domain using structs and
+enums. Creating custom types to use in your API ensures type safety: the
+compiler will make certain your functions get only values of the type each
+function expects.
+-->
+
+Vos programmes Rust peuvent maintenant décrire des concepts métier à l'aide de
```

Comment:
Is `métier` really useful here?
```suggestion
Vos programmes Rust peuvent maintenant décrire des concepts à l'aide de
```

Reply:
I don't think so, but the (english) authors has said so.
Probably to emphasise that is a tool that end-developpers can use ?

Reply:
Maybe use `concrets` then?
```suggestion
Vos programmes Rust peuvent maintenant décrire des concepts concrets à l'aide de
```